src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+        throw new Error("base-data-offset-present not supported in ttml segments");
+      }
+      if (tfFlags & 0x02) {
+        pos += 4;
+      }
+      if ((tfFlags & 0x08) === 0) {
+        throw new Error("Cannot handle ttml segments with no default_sample_duration");
+      } else {
+        pos += 4;
+      }
+      if (tfFlags && 0x10) {
+        output = concat(input.subarray(0, pos), itobe4(this.ttmlSize));
+      }
+      else{
+        throw new Error(
+          "Cannot handle ttml segments if default_sample_size_offset is absent"
+        );
+      }
+      output = concat(output, input.subarray(pos, input.length));
+      return output;
+    }
+
+    /**
+     * Process mfhd box and set segmentNumber if requested.
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    mfhd(input: Uint8Array): Uint8Array {
+      if (!this.segNr) {
+        return input;
+      }
+      const prefix = input.subarray(0, 12);
+      const segNrByte = itobe4(this.segNr);
+      return concat(prefix, segNrByte);
+    }
+
+    /**
+     * Get total duration from trun.
+     * Fix offset if self.sizeChange is non-zero.
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    trun(input: Uint8Array): Uint8Array {
+      const flags = be4toi(input, 8) & 0xFFFFFF;
+      const sampleCount = be4toi(input, 16);
+      let pos = 16;
+      let dataOffsetPresent = false;
+      // Data offset present
+      if (flags & 0x1) {
+        dataOffsetPresent = true;
+        pos += 4;
+      }
+      // First sample flags present
+      if (flags & 0x4) {
+          pos += 4;
+      }
+      const sampleDurationPresent = flags & 0x100;
+      const sampleSizePresent = flags & 0x200;
+      const sampleFlagsPresent = flags & 0x400;
+      const sampleCompTimePresent = flags & 0x800;
+      let duration = 0;
+      for (let i = 0; i<sampleCount; i++) {
+        if (sampleDurationPresent) {
+            duration += be4toi(input, pos);
+            pos += 4;
+        }
+        if (sampleSizePresent) {
+          pos += 4;
+        }
+        if (sampleFlagsPresent) {
+            pos += 4;
+      }
+        if (sampleCompTimePresent) {
+            pos += 4;
+        }
+      }
+      this.duration = duration;
+      // Modify data_offset
+      let output = input.subarray(0, 16);
+      if (dataOffsetPresent && this.sizeChange > 0) {
+        let offset = be4toi(input, 16);
+        offset += this.sizeChange;
+        output = concat(output, itobe4(offset)); // func
+      }
+      else {
+        output = concat(output, input.subarray(16,20));
+      }
+      output = concat(output, input.subarray(20, input.length));
+      return output;
+    }
+
+    /**
+     * Process sidx data and add to output.
+     * @param {Uint8Array} input
+     * @param {boolean} keepSidx
+     */
+    sidx(input: Uint8Array, patchSidx: boolean): Uint8Array {
+      let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyax&open=AWNyk0oJTmDVX2bfxyax)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyax